src/frame.js

```diff
@@ -1,6 +1,37 @@
 const EventEmitter = require('events')
 const EthereumProvider = require('ethereum-provider')
 
+function shimWeb3(provider) {
+  let loggedCurrentProvider = false
+
+  if (!window.web3) {
+    let web3Shim = { currentProvider: provider }
```

Comment:
lgtm, does assigning this twice do anything though or is it just a type?

Reply:
Not sure why it was like that, something I should have fixed after porting the approach